file-server/app.py

```diff
@@ -0,0 +1,149 @@
+import os
+import json
+from dotenv import load_dotenv
+from flask import Flask, request, jsonify
+from flask_cors import CORS
+from imgget import search_images, download_images
+import voiceovergen as vo
+import moviegen as mg
+import llm_req as llm
+import userstuff as us
+
+load_dotenv()
+
+PORT = os.environ["FILE_SERVER_PORT"]
+app = Flask(__name__)
+CORS(app)
+PALM_API_KEY = os.environ["PALMAPI"]
+
+
+@app.route("/text", methods=["POST"])
+def text():
+    """Send prompt to palm 2 API, get text. Use keywords to get images."""
+    if request.method == "POST":
+        user_id = us.generate_uuid()
+        us.generate_folder_structure(user_id)
+        data = json.loads(request.data.decode("utf-8"))
+        keywords = data["keywords"]
+        img_urls = search_images(os.environ["PEXELSAPI"], keywords)
+        download_images(img_urls, f"static/{user_id}/images")
+        imgs = os.listdir(f"./static/{user_id}/images")
+        txt = llm.generate_text(
+            "Write a 1 minute info speech in plaintext for :" + data["text"]
+        )
+        return jsonify(
+            {
+                "user_id": user_id,
+                "text": txt,
+                "image_urls": [f"/static/{user_id}/images/{image}" for image in imgs],
+            }
+        )
+
+
+@app.route("/edittext", methods=["POST"])
+def edittext():
+    """
+    Takes in a text, generates a voiceover, and creates a video with the voiceover.
+    Returns the URL of the generated video.
+    """
+    if request.method == "POST":
+        data = json.loads(request.data.decode("utf-8"))
+        edit_text = data["text"]
+        user_id = data["user_id"]
+        vo_url = vo.generate_voiceover(
+            str(edit_text), f"./static/{user_id}/audio/output_voiceover.mp3"
+        )
+        mg.create_video(
+            vo_url,
+            image_folder=f"./static/{user_id}/images",
+            output_file=f"./static/{user_id}/movie/output_video_with_audio.mp4",
+        )
+        return jsonify(
+            {"video_url": f"/static/{user_id}/movie/output_video_with_audio.mp4"}
+        )
+
+
+@app.route("/images", methods=["POST", "GET"])
+def images():
+    """
+    Takes user_id from json body and returns the urls of the images (from local storage).
+    If it doesn't exist returns an error message.
+    """
+    if request.method == "POST":
```

Comment:
This request is supposed to support POST and GET, but why is there a special case declared only for POST? Why declare it as a GET request endpoint if there is no functionality implemented for it here?

Reply:
I was using a GET version too. Removing the check and the GET from method list.